tests/store/test_sqlalchemy_store.py

```diff
@@ -317,7 +321,10 @@ def test_log_metric(self):
 
         run = self.store.get_run(run.run_uuid)
 
-        self.assertEqual(4, len(run.data.metrics))
+        sql_run_metrics = self.store._get_run(run.info.run_uuid, ViewType.ALL).metrics
+        self.assertEqual(4, len(sql_run_metrics))
+
+        self.assertEqual(3, len(run.data.metrics))
```

Comment:
Maybe add a comment here that the run_data itself should pick only the latest metric for the given key, for clarity.